tests/testthat/test-baselinenowcast.R

```diff
@@ -183,6 +193,353 @@ test_that("assert_baselinenowcast_df errors when appropriate", {
   )
 })
 
+test_that("baselinenowcast.data.frame returns the expected structure with and without strata sharing", { # nolint
+  nowcasts_df <- baselinenowcast(
+    data = covid_data,
+    max_delay = 40,
+    draws = 100,
+    strata_cols = c("age_group", "location")
+  )
+
+  expect_s3_class(nowcasts_df, "data.frame")
+  expect_s3_class(nowcasts_df, "baselinenowcast_df")
+  expect_true(all(expected_cols_nu %in% colnames(nowcasts_df)))
+  expect_identical(nowcasts_df$output_type[1], "samples")
+
+  # Check that the outputs are different for different age groups
+  expect_failure(
+    expect_equal(
+      mean(nowcasts_df$pred_count[nowcasts_df$age_group == "00+"]),
+      mean(nowcasts_df$pred_count[nowcasts_df$age_group == "60-79"]),
+      tol = 0.01
+    )
+  )
+
+  # Use strata sharing
+  # First need to remove all age groups otherwise double count
+  covid_data_age_groups <- covid_data[covid_data$age_group != "00+", ] # nolint
+  nowcasts_df2 <- baselinenowcast(
+    data = covid_data_age_groups,
+    max_delay = 40,
+    draws = 100,
+    strata_cols = c("age_group", "location"),
+    strata_sharing = c("delay", "uncertainty")
+  )
+  expect_s3_class(nowcasts_df2, "data.frame")
+  expect_s3_class(nowcasts_df2, "baselinenowcast_df")
+  expect_true(all(expected_cols_nu %in% colnames(nowcasts_df2)))
+  expect_identical(nowcasts_df2$output_type[1], "samples")
+
+  # Check that the outputs are different between with and without strata sharing
+  expect_failure(
+    expect_equal(
+      mean(nowcasts_df$pred_count[nowcasts_df$age_group == "60-79" &
+        nowcasts_df$reference_date == max(nowcasts_df$reference_date)]),
+      mean(nowcasts_df2$pred_count[nowcasts_df2$age_group == "60-79" &
+        nowcasts_df2$reference_date == max(nowcasts_df$reference_date)]),
+      tol = 0.01
+    )
+  )
+})
+
+test_that("baselinenowcast returns expected structure without errors for all different nowcast units", { # nolint
+  skip_if_not_installed("dplyr")
+  library(dplyr)
+  single_tri_data <- filter(
+    covid_data,
+    age_group == "00+"
+  ) |>
+    select(reference_date, report_date, count)
+
+  single_nowcast_df <- baselinenowcast(single_tri_data,
+    draws = 100,
+    max_delay = 40
+  )
+  expect_s3_class(single_nowcast_df, "data.frame")
```

Comment:
Herer again a custom expect function would be a nice quality of life feature as the tests are long to read